api/capi/src/nnstreamer-capi-pipeline.c

```diff
@@ -332,6 +332,56 @@ cleanup_node (gpointer data)
   g_free (e);
 }
 
+/**
+ * @brief Private function to release the pipeline resources
+ */
+static void
+cleanup_resource (gpointer data)
+{
+  pipeline_resource_s *res = data;
+
+  /* check resource type and free data */
+#if defined (__TIZEN__)
+  if (g_str_has_prefix (res->type, "tizen")) {
+    ml_tizen_release_resource (res->handle, res->type);
+  }
+#endif
+
+  g_free (res->type);
+  g_free (res);
+}
+
+/**
+ * @brief Converts predefined element in pipeline description.
+ */
+static int
+convert_element (ml_pipeline_h pipe, const gchar * description, gchar ** result)
+{
+  gchar *converted;
+  int status = ML_ERROR_NONE;
+
+  g_return_val_if_fail (pipe, ML_ERROR_INVALID_PARAMETER);
+  g_return_val_if_fail (description && result, ML_ERROR_INVALID_PARAMETER);
+
+  /* init null */
+  *result = NULL;
+
+  converted = g_strdup (description);
+
+#if defined (__TIZEN__)
+  /* convert predefined element for Tizen */
+  status = ml_tizen_convert_element (pipe, &converted);
+#endif
```

Comment:
Rather than this, define a dummy ```ml_tizen_convert_element``` in the header for ```!defined(__TIZEN__)```
E.g., in the header,
```
#if defined (__TIZEN__)
int ml_func();
#else
static int ml_func()
{
  return 0;
}
#endif
```
This makes it easier to read and easier to maintain.

Reply:
I added macros for tizen-dependent functions.